daemon/daemon_unix.go

```diff
@@ -290,6 +293,34 @@ func (daemon *Daemon) adaptContainerSettings(hostConfig *containertypes.HostConf
 	return nil
 }
 
+// When a container shares its namespace by container name,
+// use container id rather than name to identify a container.
+func (daemon *Daemon) adaptSharedNamespaceContainer(hostConfig *containertypes.HostConfig) error {
+	containerPrefix := "container:"
+	if hostConfig.PidMode.IsContainer() {
+		c, err := daemon.GetContainer(hostConfig.PidMode.Container())
```

Comment:
Maybe there is a way to call `daemon.GetContainer` once and avoid all these verbose code lines